src/main.cpp

```diff
@@ -831,6 +847,11 @@ bool AcceptToMemoryPoolWorker(CTxMemPool& pool, CValidationState &state, const C
     if (fRequireStandard && !IsStandardTx(tx, reason))
         return state.DoS(0, false, REJECT_NONSTANDARD, reason);
 
+    // Don't accept witness transactions before the final threshold passes
+    if (!GetBoolArg("-prematurewitness",false) && !tx.wit.IsNull() && !(chainActive.Tip()->nHeight + 1 >= Params().GetConsensus().SegWitHeight && IsSuperMajority(5, chainActive.Tip(), Params().GetConsensus().nMajorityRejectBlockOutdated, Params().GetConsensus()))) {
+        return state.DoS(0, false, REJECT_NONSTANDARD, "no-witness-yet");
```

Comment:
This should set "corruption possible", or else someone could maliciously add bogus witness data to non-witness transactions and blind nodes to those transactions for an extended period of time.